src/api/critter/critter.service.ts

```diff
@@ -70,6 +74,144 @@ const deleteCritter = async (critter_id: string): Promise<critter> => {
+          } : undefined,
+          lk_colour_marking_secondary_colour_idTolk_colour: m.secondary_colour ? {
+            colour: {contains: m.secondary_colour, mode: "insensitive" } 
+          } : undefined,
+          lk_marking_type: m.marking_type ? {
+            name: {contains: m.marking_type, mode: 'insensitive' }
+          } : undefined,
+          identifier: m.identifier ? String(m.identifier) : undefined
+        }
+      });
+      markings.push(...markingMatches);
+    };
+  }
+
+  const oneDay = 60 * 60 * 24 * 1000;
+  const captures = [];
+  if(body.captures) {
+    for(const c of body.captures) {
+      const capture_timestamp = c.capture_timestamp ? new Date(c.capture_timestamp) : undefined;
+      const release_timestamp = c.release_timestamp ? new Date(c.release_timestamp) : undefined;
+      const captureMatches = await prisma.capture.findMany({
+        where: {
+          capture_timestamp: capture_timestamp ? {
+            gte: capture_timestamp,
+            lt: new Date(capture_timestamp.getTime() + oneDay)
+          } : undefined,
+          release_timestamp: release_timestamp ? {
+            gte: release_timestamp,
+            lt: new Date(release_timestamp.getTime() + oneDay)
+          } : undefined,
+          location_capture_capture_location_idTolocation: formatLocationNameSearch(c.capture_location),
+          location_capture_release_location_idTolocation: formatLocationNameSearch(c.release_location)
+      }});
+      captures.push(...captureMatches);
+    }
+  }
+  
+  const mortality: mortality[] = [];
+  if(body.mortality) { 
+    const m = body.mortality;
+    const mortality_timestamp = m.mortality_timestamp ? new Date(m.mortality_timestamp) : undefined;
+    const mortality_result = await prisma.mortality.findMany({
+      where: {
+        mortality_timestamp: mortality_timestamp ? {
+          gte: mortality_timestamp,
+          lt: new Date(mortality_timestamp.getTime() + oneDay)
+        } : undefined,
+        location: formatLocationNameSearch(m.location)
+      }
+    });
+    mortality.push(...mortality_result);
+  }
+
+  let overlappingIds = Array.from(new Set([...critters.map(a => a.critter_id), ...markings.map(a => a.critter_id)]));
+  if(critters.length && markings.length) {
+    overlappingIds = intersect(critters.map(c => c.critter_id), markings.map(m => m.critter_id));
+  }
+  if(captures.length) {
+    overlappingIds = intersect(captures.map(c => c.critter_id), overlappingIds);
+  }
+  if(mortality.length) {
+    overlappingIds = intersect(mortality.map(m => m.critter_id), overlappingIds);
+  }
+
+  const detailedCritters = await prisma.critter.findMany({
+    ...(body.detail && formattedCritterInclude),
+    where: {
+      critter_id: { in: (critters.length == 1 ? critters.map(c => c.critter_id) : overlappingIds) }
+    }
+  });
+  return body.detail ? detailedCritters.map(c => CritterResponseSchema.parse(c)) : detailedCritters;
+}
+
+const getTableDataTypes = async () => {
```

Comment:
Maybe move this to helper functions and pass the table name to function.
I think prisma has a type for model name

Reply:
move to access router